projects/plugins/crm/src/automation/commons/conditions/contacts/class-contact-transitional-status.php

```diff
@@ -54,19 +54,11 @@ public function __construct( array $step_data ) {
 	 *
 	 * @throws Automation_Exception If an invalid operator is encountered.
 	 */
-	public function execute( Data_Type $data ) {
-		$this->validate( $data );
-
+	protected function execute( Data_Type $data ) {
+		/** @var Contact $contact */
 		$contact          = $data->get_data();
 		$previous_contact = $data->get_previous_data();
 
-		if ( ! $previous_contact instanceof Contact ) {
-			$this->logger->log( 'Invalid previous contact status transitional data' );
-			$this->condition_met = false;
-
-			return;
-		}
-
```

Comment:
By design, the previous data is a Data_Type too. We don't need to validate anything here.